plugins/outputs/signalfx/signalfx.go

```diff
@@ -0,0 +1,230 @@
+	SignalFxRealm      string   `toml:"signalfx_realm"`
+	IngestURL          string   `toml:"ingest_url"`
+	IncludedEventNames []string `toml:"included_event_names"`
+
+	includedEventSet map[string]bool
+	client           dpsink.Sink
+
+	ctx    context.Context
+	cancel context.CancelFunc
+	wg     sync.WaitGroup
+}
+
+var sampleConfig = `
+    ## SignalFx Org Access Token
+    access_token = "my-secret-token"
+
+    ## The SignalFx realm that your organization resides in
+    signalfx_realm = us9  # Required if ingest_url is not set
+
+    ## You can optionally provide a custom ingest url instead of the
+    ## signalfx_realm option above if you are using a gateway or proxy
+    ## instance.  This option takes precident over signalfx_realm.
+    ingest_url = https://my-custom-ingest/
+
+    ## Event typed metrics are omitted by default,
+    ## If you require an event typed metric you must specify the
+    ## metric name in the following list.
+    included_event_names = ["plugin.metric_name", ""]
+`
+
+// GetMetricType returns the equivalent telegraf ValueType for a signalfx metric type
+func GetMetricType(mtype telegraf.ValueType) (metricType datapoint.MetricType) {
+	switch mtype {
+	case telegraf.Counter:
+		metricType = datapoint.Counter
+	case telegraf.Gauge:
+		metricType = datapoint.Gauge
+	case telegraf.Summary:
+		metricType = datapoint.Gauge
+	case telegraf.Histogram:
+		metricType = datapoint.Gauge
+	case telegraf.Untyped:
+		metricType = datapoint.Gauge
+	default:
+		metricType = datapoint.Gauge
+	}
+	return metricType
+}
+
+// NewSignalFx - returns a new context for the SignalFx output plugin
+func NewSignalFx() *SignalFx {
+	ctx, cancel := context.WithCancel(context.Background())
+	return &SignalFx{
+		AccessToken:        "",
+		SignalFxRealm:      "",
+		IngestURL:          "",
+		IncludedEventNames: []string{""},
+		ctx:                ctx,
+		cancel:             cancel,
+	}
+}
+
+// Description returns a description for the plugin
+func (s *SignalFx) Description() string {
+	return "Send metrics and events to SignalFx"
+}
+
+// SampleConfig returns the sample configuration for the plugin
+func (s *SignalFx) SampleConfig() string {
+	return sampleConfig
+}
+
+// Connect establishes a connection to SignalFx
+func (s *SignalFx) Connect() error {
+	// Make a connection to the URL here
+	client := sfxclient.NewHTTPSink()
+	client.AuthToken = s.AccessToken
+
+	if s.IngestURL != "" {
+		client.DatapointEndpoint = realm.DatapointEndpointForIngestURL(s.IngestURL)
+		client.EventEndpoint = realm.EventEndpointForIngestURL(s.IngestURL)
+	} else if s.SignalFxRealm != "" {
+		client.DatapointEndpoint = realm.DatapointEndpointForRealm(s.SignalFxRealm)
+		client.EventEndpoint = realm.EventEndpointForRealm(s.SignalFxRealm)
+	} else {
+		return errors.New("signalfx_realm or ingest_url must be configured")
+	}
+
+	s.client = client
+
+	return nil
+}
+
+// Close closes the connection to SignalFx
+func (s *SignalFx) Close() error {
+	s.cancel()
```

Comment:
Changed it to close any idle connections.

Reply:
fyi there should be zero non-idle connections, or something went really wrong.

Reply:
If the HTTP client has persistent connections enabled then there could be idle connections from previous requests, so this will make sure they get stopped with the plugin.